appservice/src/deploy/localGitDeploy.ts

```diff
@@ -32,30 +33,63 @@ export async function localGitDeploy(client: SiteClient, fsPath: string): Promis
             await ext.ui.showWarningMessage(message, { modal: true }, deployAnyway, DialogResponses.cancel);
         }
         await verifyNoRunFromPackageSetting(client);
-        // tslint:disable-next-line:no-floating-promises
-        localGit.push(remote, 'HEAD:master');
+
+        // wrap two async actions in a promise because both need to run and finish simaltaneously
+        await new Promise((resolve: () => void, reject: (err: IParsedError) => void): void => {
+            ext.outputChannel.appendLine(formatDeployLog(client, (localize('localGitDeploy', `Deploying Local Git repository to "${client.fullName}"...`))));
+            localGit.push(remote, 'HEAD:master').catch(async (error) => {
+                tokenSource.cancel();
+                // tslint:disable-next-line:no-unsafe-any
+                const parsedError: IParsedError = parseAndRemoveCredentialsFromError(error);
+
+                // if the push fails, check if it's due to diverging history to offer force push
+                if (parsedError.message.indexOf('Updates were rejected because the remote contains work that you do') >= 0) {
```

Comment:
I don't trust this error message to stay the same - or even to stay in english. I think we should be checking a code or something else. If there's no code, maybe we should be checking for diverging history before a git push

Reply:
```
"To https://naturins-22-error-03.scm.azurewebsites.net:443/naturins-22-error-03.git
 ! [rejected]        HEAD -> master (fetch first)
error: failed to push some refs to 'https://$naturins-22-error-03:***@naturins-22-error-03.scm.azurewebsites.net:443/naturins-22-error-03.git'
hint: Updates were rejected because the remote contains work that you do
hint: not have locally. This is usually caused by another repository pushing
hint: to the same ref. You may want to first integrate the remote changes
hint: (e.g., 'git pull ...') before pushing again.
hint: See the 'Note about fast-forwards' in 'git push --help' for details.
```

This is the error that comes back.  I looked into trying to figure out diverging history prior to a push, but it involves fetching the entire remote and then comparing it locally.  It wasn't that complicated, but it took a bit longer to do since we'd do it with every push.

I honestly don't know how often people use Force Push (should have gathered telemetry, stupid past Nathan 😞) in the first place or if we should even encourage them to.  Seems kind of like overriding their old deployment might not be something we would want to allow since it's kind of a "one button deploy" scenario.